### DISCUSSION TITLE
Consider adding error-codes

### DISCUSSION THREAD

**Discussion by jaybosamiya** [verus-lang/verus#641] [General]
It would be useful for us to have each error reported by Verus come with an error code, similar to Rust (see https://doc.rust-lang.org/stable/error_codes/error-index.html).

This would allow users who face an error to quickly look-up more details, or easily jump to relevant part of code in Verus, and so on. Also, it would help create a point for documentation to build on, for example get more hints on ways to fix up the error and such.

---
Opening this as a separate discussion from https://github.com/verus-lang/verus/discussions/387 since that one focuses on the specific error messages itself, but this is about updating all error codes.

**Comment by jaybosamiya** [comment]
My suggestion wrt the format itself would be to just use the format that Rust uses, i.e. `E[0-9]{4}` (eg: `E0001`, `E0007`, ...), but could start from a higher number, if we are worried about collisions with Rust's error codes.

Another alternative is to use `V[0-9]{4}` to indicate Verus.

**Comment by utaal** [comment]
[retreat 2025]

@amaurremi: it would be helpful if error codes led to links with further explanation of the error, especially in a context where engineers are being onboarded.
